src/zenml/orchestrators/step_runner.py

```diff
@@ -235,49 +231,138 @@ def _load_input_artifact(
 
         Returns:
             The artifact value.
-
-        Raises:
-            RuntimeError: If the artifact has no materializer.
         """
-        # Skip materialization for BaseArtifact and its subtypes.
+        # Skip materialization for `UnmaterializedArtifact`.
+        if data_type == UnmaterializedArtifact:
+            return UnmaterializedArtifact.parse_obj(artifact.dict())
```

Comment:
```suggestion
            return UnmaterializedArtifact.parse_obj(artifact)
```

This is a pydantic shorthand for the same thing, they call `model.dict()` internally